src/SprykerSdk/Zed/AppSdk/Business/Validator/AsyncApi/Validator/DuplicateMessageValidator.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\Zed\AppSdk\Business\Validator\AsyncApi\Validator;
+
+use Generated\Shared\Transfer\MessageTransfer;
+use Generated\Shared\Transfer\ValidateResponseTransfer;
+use SprykerSdk\Zed\AppSdk\AppSdkConfig;
+use SprykerSdk\Zed\AppSdk\Business\Validator\FileValidatorInterface;
+
+class DuplicateMessageValidator implements FileValidatorInterface
+{
+    /**
+     * @var \SprykerSdk\Zed\AppSdk\AppSdkConfig
+     */
+    protected AppSdkConfig $config;
+
+    /**
+     * @param \SprykerSdk\Zed\AppSdk\AppSdkConfig $config
+     */
+    public function __construct(AppSdkConfig $config)
+    {
+        $this->config = $config;
+    }
+
+    /**
+     * @param array $data
+     * @param string $fileName
+     * @param \Generated\Shared\Transfer\ValidateResponseTransfer $validateResponseTransfer
+     * @param array|null $context
+     *
+     * @return \Generated\Shared\Transfer\ValidateResponseTransfer
+     */
+    public function validate(
+        array $data,
+        string $fileName,
+        ValidateResponseTransfer $validateResponseTransfer,
+        ?array $context = null
+    ): ValidateResponseTransfer {
+        if (isset($data['components']['messages'])) {
```

Comment:
You can avoid if/else when you use early return which is the preferred way of Spryker.

```
if (!isset($data['components']['messages'])) {
    // add error
    return $validateResponseTransfer; 
}
```

Your approach adds too much nesting of function calls `if / foreach / if`. There should never be more than two.